ginivision/src/main/java/net/gini/android/vision/camera/CameraActivity.java

```diff
@@ -395,28 +477,44 @@ private void startOnboardingActivity() {
     @Override
     public void onDocumentAvailable(@NonNull Document document) {
         mDocument = document;
-        // Start ReviewActivity
-        mReviewDocumentActivityIntent.putExtra(ReviewActivity.EXTRA_IN_DOCUMENT, document);
-        mReviewDocumentActivityIntent.putExtra(EXTRA_IN_ANALYSIS_ACTIVITY, mAnalyzeDocumentActivityIntent);
-        mReviewDocumentActivityIntent.putExtra(ReviewActivity.EXTRA_IN_BACK_BUTTON_SHOULD_CLOSE_LIBRARY,
-                mBackButtonShouldCloseLibrary);
-        startActivityForResult(mReviewDocumentActivityIntent, REVIEW_DOCUMENT_REQUEST);
+        if (mDocument.isReviewable()) {
+            // Start ReviewActivity
+            final Intent reviewIntent = new Intent(mReviewDocumentActivityIntent);
+            reviewIntent.putExtra(ReviewActivity.EXTRA_IN_DOCUMENT, document);
+            reviewIntent.putExtra(EXTRA_IN_ANALYSIS_ACTIVITY,
+                    mAnalyzeDocumentActivityIntent);
+            reviewIntent.putExtra(
+                    ReviewActivity.EXTRA_IN_BACK_BUTTON_SHOULD_CLOSE_LIBRARY,
+                    mBackButtonShouldCloseLibrary);
+            startActivityForResult(reviewIntent, REVIEW_DOCUMENT_REQUEST);
+        } else {
+            // Start AnalysisActivity
+            final Intent analysisIntent = new Intent(mAnalyzeDocumentActivityIntent);
+            analysisIntent.putExtra(AnalysisActivity.EXTRA_IN_DOCUMENT, document);
+            startActivityForResult(analysisIntent, ANALYSE_DOCUMENT_REQUEST);
+        }
     }
 
     @Override
     public void onError(@NonNull GiniVisionError error) {
-        Intent result = new Intent();
-        result.putExtra(EXTRA_OUT_ERROR, error);
-        setResult(RESULT_ERROR, result);
-        finish();
+        if (error.getErrorCode() == GiniVisionError.ErrorCode.DOCUMENT_IMPORT) {
```

Comment:
Document import error shown in a snackbar as in the Analysis Screen.

Reply:
I'll move the showing of the snackbar for document import errors to the fragment so that it works the same for the Component API.